components/Wallet/Send.tsx

```diff
@@ -0,0 +1,251 @@
+import { Message } from '@glif/filecoin-message'
+import { FilecoinNumber, BigNumber } from '@glif/filecoin-number'
+import { useWallet, useWalletProvider } from '@glif/wallet-provider-react'
+import {
+  getMaxAffordableFee,
+  getMaxGasFee,
+  getTotalAmount,
+  useGetGasParams,
+  InputV2,
+  Dialog,
+  ErrorBox,
+  ShadowBox,
+  StandardBox,
+  Transaction,
+  LoginOption
+} from '@glif/react-components'
+
+import { navigate } from '../../utils/urlParams'
+import { PAGE } from '../../constants'
+import { logger } from '../../logger'
+
+export const Send = () => {
+  const router = useRouter()
+  const wallet = useWallet()
+  const { loginOption, walletProvider, walletError, resetWalletError } =
+    useWalletProvider()
+
+  // Input states
+  const [toAddress, setToAddress] = useState<string>('')
+  const [value, setValue] = useState<FilecoinNumber | null>(null)
+  const [params, setParams] = useState<string>('')
+  const [txFee, setTxFee] = useState<FilecoinNumber | null>(null)
+  const [isToAddressValid, setIsToAddressValid] = useState<boolean>(false)
+  const [isValueValid, setIsValueValid] = useState<boolean>(false)
+  const [isParamsValid, setIsParamsValid] = useState<boolean>(false)
+  const [isTxFeeValid, setIsTxFeeValid] = useState<boolean>(false)
+  const inputsValid =
+    isToAddressValid && isValueValid && isParamsValid && isTxFeeValid
+
+  // Sending states
+  const [isSending, setIsSending] = useState<boolean>(false)
+  const [sendError, setSendError] = useState<Error | null>(null)
+
+  // Placeholder message for getting gas params
+  const [message, setMessage] = useState<Message | null>(null)
+
+  // Prevent redundant updates to message so that we don't
+  // invoke the useGetGasParams hook more than necessary
+  const setMessageIfChanged = () => {
+    if (!isToAddressValid || !isValueValid || !isParamsValid) {
+      setMessage(null)
+      return
+    }
+    if (
+      !message ||
+      message.to !== toAddress ||
+      message.value.toString() !== value.toAttoFil() ||
+      message.params !== params
+    )
+      setMessage(
+        new Message({
+          to: toAddress,
+          from: wallet.address,
+          nonce: 0,
+          value: value.toAttoFil(),
+          method: 0,
+          params: params,
+          gasPremium: '0',
+          gasFeeCap: '0',
+          gasLimit: 0
+        })
+      )
+  }
+
+  // Max transaction fee used for getting gas params. Will be
+  // null until the user manually changes the transaction fee.
+  const [maxFee, setMaxFee] = useState<FilecoinNumber | null>(null)
+
+  // Load gas parameters when message or max fee changes
+  const {
+    gasParams,
+    loading: gasParamsLoading,
+    error: gasParamsError
+  } = useGetGasParams(walletProvider, message, maxFee)
+
+  // Calculate max affordable fee (balance minus value)
+  const maxAffordableFee = useMemo<FilecoinNumber | null>(() => {
+    return value ? getMaxAffordableFee(wallet.balance, value) : null
```

Comment:
Just to be extra cautious (i dont think this is happening anywhere), but we shouldn't be setting the maxFee to maxAffordableFee ever - weird consequences will ensue

Reply:
Ok, noted!